src/core/router/router.blueprint.tsx

```diff
@@ -172,6 +173,20 @@ export const blueprint: RouteObject[] = [
           },
         ],
       },
+      {
+        path: 'contracts',
+        loader: async () => {
+          const requests = [userOffers({ page: 1, limit: 5 }), identities()];
+          const [offers, currentIdentities] = await Promise.all(requests);
+          return { offers, currentIdentities };
```

Comment:
You don't have to get identities in loader, just pass your component to Protect()